src/terminal/char_grid/cursor.rs

```diff
@@ -24,10 +24,20 @@ pub struct Cursor {
     pub coords: Coords,
     pub style: Styles,
     pub text_style: Styles,
+    config: Config,
```

Comment:
Instead of duplicating `Config` here, do you think you could add an argument to every function that needs that info on `Cursor`? There's a 1-to-1 relationship between `Cursor` and a `CharGrid` so there's no point duplicating this info.